File: Data Structures & Algorithms/maximum-product-subarray/submission-6.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        currMax, currMin = 1,1
        res = nums[0]

        for n in nums:
            if n==0:
                currMax, currMin = 1,1

            tmp = currMax
            currMax = max(n, currMax*n, currMin*n)
            currMin = min(n, tmp*n, currMin*n)
            res = max(res, currMax)
        
        return res

File: Data Structures & Algorithms/reverse-string/submission-6.cs
public class Solution {
    public void ReverseString(char[] s) {
        char[] tmp = new char[s.Length];
        int n = s.Length;
        int mid = n/2;

        for (int i=0; i<mid; i++) {
            char tmpX = s[i];
            s[i] = s[n-1-i];
            s[n-1-i] = tmpX;
        }

    }
}